### ISSUE TITLE
Facility File Testing

### ISSUE THREAD

**Issue by anthonycpichardo** [lzim/teampsd#1667]
Say Guides: 
 (Red): https://github.com/lzim/mtl/blob/feature-directory_scan/red/part1/part1_facilitator/mtl_red_part_1_say.md

 (Blue): https://github.com/lzim/mtl/blob/feature-directory_scan/blue/session02/s02_facilitator/mtl_session02_say.md

See Guides:
(Red): https://github.com/lzim/mtl/blob/feature-directory_scan/red/part1/part1_learner/mtl_red_part_1_see.md

(Blue): https://github.com/lzim/mtl/blob/feature-directory_scan/blue/session02/s02_learner/mtl_session02_see.md

**facility.xlsb file** :
- [x] **divisionname**  filter is visible on the graphs in the vizEnc, vizdiag, vizHF and vizMeas tabs
- [x] There are no unusable buttons on the graphs-- all buttons are active
- [x] VVC filters are working and correctly drill down the data on the vizenc tab
- [x]  The "Hidden" tab is not visible

**detail_facilitate:** (for _MTL_ Manual - Facilitator section).
- [x] Documentation for unhiding the **Control** Tab and Refreshing the data is clear and easy to follow

**Comment by lzim** [comment]
@anazariz and @anthonycpichardo 

Did @jamesmrollins and @dkngenda receive their facility testing files earlier today okay?
I haven't seen updates for each step on here (yet).

_We need sign-offs on each step of QA, likely in the future these should each be separate cards with assignees, dependencies and estimates._
-[ ] DEV QA - Anthony & Ash (sign-off)
-[ ] TEST QA - Stacey Documentation (sign-off)
-[ ] TEST QA - James and David walk through guides and facility.xlsxb file (sign-off).

**Comment by anthonycpichardo** [comment]
I agree @lzim that in the future we would have a card for each test step.

Ash and I are wrapping up dev testing now and will have the test files ready shortly for folks to be able to test in the morning. 

The data query took more time to run than scoped.

FYI: To ensure we have the data ready for our sites we will be running the propagation for the wave 1 sites prior to propagating all of the facilities

**Comment by anazariz** [comment]
@anazariz Test Results: vizDiag chart did not size correctly on workbook load - fixed.

**Comment by anthonycpichardo** [comment]
Testing results - 
- [x]  divisionname filter is visible on the graphs in the vizEnc, vizdiag, vizHF and vizMeas tabs - OK
- [x]  There are no unusable buttons on the graphs-- all buttons are active - OK
- [x]  VVC filters are working and correctly drill down the data on the vizenc tab - OK
- [x]  The "Hidden" tab is not visible - OK.
Feedback: 
Need to remove the pivotChart Fields settings from being open on launch.

**Comment by lijenn** [comment]
@anthonycpichardo @anazariz I'm still in the process of testing the facility file, I can recall the decision but did we want the Control tab to be unhidden on default? 

@dkngenda @jamesmrollins the documentation has been updated to include the Control tab documentation. Please see updated links to the SEE/SAY guide as they are in a feature branch in the lzim/mtl repo and not in the master branch yet.

Please go ahead and start testing the facility file now, thank you!

**Comment by anthonycpichardo** [comment]
Good question @lijenn, I think it makes sense for it to be hidden as users won't need to use it in 99% of cases. The 1% being if for some reason they want to refresh the data?

**Comment by dkngenda** [comment]
@anthonycpichardo @anazariz graphs are appearing on top of some tables, is that okay? - looks like they can be moved around

**Comment by anthonycpichardo** [comment]
Yes they are intentionally on top of the tables

**Comment by anazariz** [comment]
@anthony, re your feeback, "need to remove the pivotChart Fields settings from being open on launch", yes, I will remove them prior to propagation.

**Comment by anazariz** [comment]
@lzim @staceypark @lijenn  @dkngenda @jamesmrollins 
folks, please move your tested (or untested) files to either completed\<current date> or did not complete\<current date>. That way @anthonycpichardo and I will know the testing that still remains if we are to propagate the file(s) today. 
Thanks.

**Comment by dkngenda** [comment]
Testing results

- [x]  divisionname filter is visible on the graphs in the vizEnc, vizdiag, vizHF and vizMeas tabs - OK
- [x]  There are no unusable buttons on the graphs-- all buttons are active - OK
- [x]  VVC filters are working and correctly drill down the data on the vizenc tab - OK
- [x]  The "Hidden" tab is not visible

**Comment by lijenn** [comment]
Testing results: 
 - [x] divisionname filter is visible on the graphs in the vizEnc, vizdiag, vizHF and vizMeas tabs - OK
 - [x] There are no unusable buttons on the graphs-- all buttons are active - OK
 - [x] VVC filters are working and correctly drill down the data on the vizenc tab - OK
 - [x]  The "Hidden" tab is not visible - OK

Some issues I did see, but able to recreate:
- Same issue that Stacey previously mentioned where the lines in the graph extend into the legend, here its in VizDiag tab and appeared when I randomly filtered through the Division Name. 
(FYI I tried to recreate this a 2nd time but it seems to be fine now? Was unable to recreate after I clicked 'Reset Pivot Chart' and worked fine afterwards).
![image](https://user-images.githubusercontent.com/59668647/106672152-76d90080-6564-11eb-9318-031144cae7f3.png)

- When all filters are selected and filtered (divisionname, locationname, and VVC - selected to VVC) in VizEnc tab, nothing appears in the graph. But when VVC is selected to non-VVC, the graph appears.
Also tried to recreate this but now it seems to be fine?
![image](https://user-images.githubusercontent.com/59668647/106672233-95d79280-6564-11eb-8baf-768bbc213a69.png)

**Comment by anthonycpichardo** [comment]
Interesting that both of the errors seem to be inconsistent

**Comment by dkngenda** [comment]
I got this  message when going through the facilitate.xlb
![image](https://user-images.githubusercontent.com/66848852/106673592-d2fd4e00-657f-11eb-8d7d-61f5c402dcf4.png)

**Comment by anthonycpichardo** [comment]
Was this the file that Ash pointed you to in the test folder? The workbook being mentioned in the alert is not the same as the test file.

Select no if you still have the dialog open.

**Comment by dkngenda** [comment]
My bad! I was just going through the facilitate manual clicking around to follow the instructions

**Comment by anthonycpichardo** [comment]
ah I see. You got this error when navigating to 605? I will test this as well.

**Comment by staceypark** [comment]
@anthonycpichardo @anazariz 

**facility.xlsb file** :
- [x] **divisionname**  filter is visible on the graphs in the vizEnc, vizdiag, vizHF and vizMeas tabs
- [x] There are no unusable buttons on the graphs-- all buttons are active
- [x] VVC filters are working and correctly drill down the data on the vizenc tab
- [x]  The "Hidden" tab is not visible

The above items were all fine.
An additional point though, that isn't a bug, so much as a usability issue. Some of the filters have "Select Multiple Items" pre-selected and shows the checkboxes which you can click and unclick. Some do not. I think consistency across the filters on how they work is key for reducing confusion.

Between the two options, I would lean towards showing the checkboxes with the pre-selected "select multiple items". Less confusing on how to use the filters to me.

from VizDiag division name
![image](https://user-images.githubusercontent.com/31089501/106677039-d935ff00-656c-11eb-8879-ab64cbeaead5.png)

from VizEnc location name
![image](https://user-images.githubusercontent.com/31089501/106677080-f539a080-656c-11eb-81ce-fcf3eba5d4fe.png)

**Comment by anthonycpichardo** [comment]
Good feedback. I think we would need to explore how to do that.

**Comment by dkngenda** [comment]
@lijenn @staceypark 
Say (Red) guide
- Section 2 bullet 2 is not entirely clear to me. Are we referring to the folder on the splash page?
- There is a reference to these tabs " countDiag, countEnc, countHF, and countMeas;" but I din't see them in the facility file.

**Comment by anthonycpichardo** [comment]
@lzim We have received feedback from all users other than James who is unable to access our Shared Drive, the requirements outlined on the testing checklist above have been met.

Recommending that we begin the propagation for the wave 1 sites.

**Comment by staceypark** [comment]
Ok we will take a look
> @lijenn @staceypark
> Say (Red) guide
> 
> * Section 2 bullet 2 is not entirely clear to me. Are we referring to the folder on the splash page?
> * There is a reference to these tabs " countDiag, countEnc, countHF, and countMeas;" but I din't see them in the facility file.

**Comment by staceypark** [comment]
Is it as simple as pre-clicking those checkboxes for "select multiple items" pre-propagation?

> Good feedback. I think we would need to explore how to do that.

**Comment by anthonycpichardo** [comment]
> Is it as simple as pre-clicking those checkboxes for "select multiple items" pre-propagation?
> 
> > Good feedback. I think we would need to explore how to do that.

I can test that, but I'm not sure how that would play with the reset button.

If its that simple that would be great.

**Comment by anthonycpichardo** [comment]
> > Is it as simple as pre-clicking those checkboxes for "select multiple items" pre-propagation?
> > > Good feedback. I think we would need to explore how to do that.
> 
> I can test that, but I'm not sure how that would play with the reset button.
> 
> If its that simple that would be great.

That seems to have worked, I'll try that.

**Comment by jamesmrollins** [comment]
@lijenn with regard to the Red and Blue Say guides, I have the following comments.  I did not have enough time to resolve issues with Excel, run very slow queries and check **all** the guides.  However, the key ask above, " Documentation for unhiding the Control Tab and Refreshing the data is clear and easy to follow" has not been achieved.  Since the guides all seem derivative, we likely can assume the changes would need to be made across all of them.  

I also have a few more comments as well.  These may be hard to follow, but I am willing to review them in person with you.

### mtl_red_part_1_say.md

1. bullet 3 states: "A web version of Excel will open on your screen, showing your facility's data in charts."

What actually happened:  A directory in BISL is displayed which indicates a 663_facility.xlsb file is displayed.

2. bullet 2 states: "If you do not see a team folder, request a team folder above."  

Observation:  It is not clear what "folder above" means. On first entry into the system, there wouldn't be a team folder, or is someone else creating it?  Is it created when the team goes through co-planning?

3.  Time Stamp - Team Learning Priorities - Script for Session 2, table needs to be fixed.

4.  "If the data is not loading or showing incorrect facilities/clinics, here's how to refresh" refresh procedure is not consistent with actual dialogue in spreadsheet.  

-  The second bullet indicates a "control" selection will be in the "Unhide" menu.  The only two selections presented in the spreadsheet were "drop down" and "Hidden."

- I think the problem is the first bullet instruction says, "To refresh your facility file's data, right click on any tab at the bottom of the Excel spreadsheet and select **Unhide..."  I think the user must actually navigate to the "Control" worksheet tab, select a facility in cell C2 on the top of the worksheet and click the "Refresh" button on the top of the page. Revealing the hidden tab is a function of the Propogate button according to the instructions on the worksheet. 

5.  We have been working in the beginning of the Say guide in the splashpage.  Then we transition back to BISL and then to the Data UI file.  Consider a more solid transition.  

-  Add "Go to BISL (or SharePoint Online).  Scroll to your team folder at the bottom of the page. Open the data_ui folder and open your data_ui file in Excel."

6.  Regarding #4. Go to the ClinicSelection tab. Use columns C-H . . . 

-  Third bullet says, "Note that this will pull all clinics used in the last two years (including de-activated clinics: denoted by "ZZ"). You can see the de-activated clinics in column I."  Consider, "Note that this will pull all clinics used in the last two years, including de-activated clinics (denoted by a "ZZ" prefix). You can also see the de-activated clinic's inactivation date in column I."

- Forth bullet says, "Follow the instructions in Box A2."  Consider, "Follow the instructions in cell A2." Or - draw a box around the cell and keep the "box" language.  The word "cell" is used in the two bullets that follow bullet 4

Note:  Consider color coding the buttons in cells B3 through B6.  It took me a few minutes to find the "add all" button.  Also consider capitalizing the names of the buttons in the instructions in cell A2. Also, consider left-justifying the cell and bulletizing the instructions in A2, it will be easier to read.


### mtl_session02_say.md

1.  Under #2, bullet 2, says, "Team files are in team folders. If you do not see a team folder, request a team folder above."  Where is above?

2.  Under #2, is says, "There are two types of View options at the bottom of the spreadsheet. __________________ (Team Lead's name) Will you please click through them, so we can see what each displays?"  What view options? Do we mean to say, "There are two views available in each worksheet, a graph and a table." Or, "There are data view options available in the tabs at the bottom of the spreadsheet." 

3. Under heading "All data is sortable and filterable so that the team can transparently see how the charting you do all day becomes VA data." 

- Regarding # 2, "For each "viz" chart, you can filter on location and clinic name (upper left) to zoom in to the clinic or VISTA clinic/"grid". (During intersession prep facilitator should fill-in the rankings of the highest priority viz that should be emphasized during the team meeting. Follow the team's lead "live" during the meeting, but be prepared.)."  

- The options on the chart are "divisioname" and "locationname" and are inconsistent with " . . . location and clinic name (upper left) . . ." 

4. Under "If the data is not loading or showing  . . ." this needs to be corrected.

5. Under heading, "To create a fresh data UI file, go through Steps #3-5."

- Regarding #4. Go to the ClinicSelection tab. Use columns C-H . . . 

- Third bullet says, "Note that this will pull all clinics used in the last two years (including de-activated clinics: denoted by "ZZ"). You can see the de-activated clinics in column I."  Consider, "Note that this will pull all clinics used in the last two years, including de-activated clinics (denoted by a "ZZ" prefix). You can also see the de-activated clinic's inactivation date in column I."

- Forth bullet says, "Follow the instructions in Box A2."  Consider, "Follow the instructions in cell A2." Or - draw a box around the cell and keep the "box" language.  The word "cell" is used in the two bullets that follow bullet 4

6.   Time Stamp - Team Learning Priorities - Script for Session 2, table needs to be fixed.

**I also recommend we go through a lint process together after I delete the heading punctuation rule.  It may speed up our editing process if I can share my lessons learned with you.**

**Comment by staceypark** [comment]
@anthonycpichardo 
@lzim confirmed propagation for wave 1. She will be checking Fresno at 5:30am
> @lzim We have received feedback from all users stating the requirements outlined on the testing checklist above have been met.
> 
> Recommending that we begin the propagation for the wave 1 sites.

**Comment by anthonycpichardo** [comment]
Great will be running it now and verifying that every thing is peachy keen

**Comment by lzim** [comment]
@anthonycpichardo @anazariz 
- Edit to @staceypark  above 
- I'll turn to prep Fresno after 5:30PM **Tonight** (not 5:30AM tomorrow).

I will not have time for any other testing other than my preparations for tomorrow.

Even if we don't have separate cards, I still do want us to sign off on the QA roles for each step (from my post yesterday):
> We need sign-offs on each step of QA, likely in the future these should each be separate cards with assignees, dependencies > and estimates.

 > DEV QA - Anthony & Ash (sign-off)
 > TEST QA - Stacey Documentation (sign-off)
 > TEST QA - James and David walk through guides and facility.xlsxb file (sign-off).

**Comment by anthonycpichardo** [comment]
Thanks for the update, the latest facility file should be available for 570 @lzim

**Comment by anthonycpichardo** [comment]
Facility File Feedback to be implemented:
- font size is too small on the buttons for the viz

**Comment by anthonycpichardo** [comment]
Documentation being tracked under #1678

**Comment by hnorthcraft** [comment]
#adjudicate_memo per comment above, efforts trackd to other card, marking as dup
